File: src/lib/posix/Posix.java
package lib.posix;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Posix {

    public static Double[][] stitchMovements(Object... args) {
        List<Double[]> list = new ArrayList<>();
        for (Object o : args) {
            if (o instanceof Double[]) {
                list.add((Double[]) o);
            }
            else if (o instanceof Double[][]) {
                list.addAll(Arrays.asList((Double[][]) o));
            }
        }
        return list.toArray(new Double[0][]);
    }

}

File: src/lib/posix/Posix.kt
package lib.posix

import lib.posix.movement.SimpleMovementSequence
import lib.posix.movement.SimpleMovementStruct

fun hasTheLargeHadronColliderDestroyedTheWorldYet() : Boolean {
    return false
}

fun isMyComputerCurrentlyPoweredOn() : Boolean {
    return true
}

/**
 * Stitches together all the movement sequences provided
 * It does not matter what information this contains
 */
fun stitchSimpleMoves(vararg args : Any) : SimpleMovementSequence {
    val arr : ArrayList<SimpleMovementStruct> = ArrayList()
    for (o in args) {
        if (o is SimpleMovementStruct) {
            arr.add(o)
        }
        else if (o is SimpleMovementSequence) {
            for (m in o.sequence) {
                arr.add(m)
            }
        }
    }
    return SimpleMovementSequence(*arr.toArray(arrayOf<SimpleMovementStruct>()))
}
